packages/server/src/controllers/UsersController.js

```diff
@@ -4,7 +4,135 @@ import UsersValidator from '../validators/UsersValidator'
 
 const { User, UserAddress } = Models
 
+// temp dummy models
+let Users = { }
+let UserEmails = { }
+let UserSessions = { }
+
+
 export default {
+  async details(req, res) {
+    let body = {
+      "emailExists": false,
+      "emailVerified": false,
+      "addressVerified": false,
+      "notificationsEnabled": false
+    }
+    Object.assign(body, Users[req.params.address])
+    res.send(body)
+  },
+
+
+  sessions: {
+    async create(req, res) {
+      if (!(req.params.address in Users)) {
+        Users[req.params.address] = {}
+      }
+      Users[req.params.address]['addressVerified'] = true
+      if (!(req.params.address in UserSessions)) {
+        UserSessions[req.params.address] = {}
+      }
+      UserSessions[req.params.address][req.session.id] = true
+      let body = {
+        "authenticated": true
+      }
+      res.send(body)
+    },
+
+    async authenticate(req,res,next) {
+      if (
+        !(req.params.address in UserSessions) ||
+        !(req.session.id in UserSessions[req.params.address])
+      ) {
+        // exception for new emails from anj
+        if (req.path.endsWith("/email") && !(req.params.address in Users)) {
+          return next()
+        }
+        const errors = [{access: `Unauthorized, please authenticate at /users/${req.params.address}/sessions`}]
+        throw HttpError._403({ errors })
+      }
+      next()
+    },
+    
+    async deleteCurrent(req, res) {
+      delete UserSessions[req.params.address][req.session.id]
+      let body = {
+        "deleted": true
+      }
+      res.send(body)
+    },
+    
+    async deleteAll(req, res) {
+      delete UserSessions[req.params.address]
+      let body = {
+        "deleted": true
+      }
+      res.send(body)
+    },
+  },
+
+
+  email: {
+    async get(req, res) {
+      let body = {
+        "email": UserEmails[req.params.address],
+      }
+      res.send(body)
+    },
+
+    async change(req, res) {
+      UserEmails[req.params.address] = req.body.email
+      if (!(req.params.address in Users)) {
+        Users[req.params.address] = {}
+      }
+      Users[req.params.address]['emailExists'] = true
+      Users[req.params.address]['notificationsEnabled'] = true
+      let body = {
+        "email": req.body.email,
+        "sent": true
+      }
+      res.send(body)
+    },
+    
+    async verify(req, res) {
+      Users[req.params.address]['emailVerified'] = true
+      let body = {
+        "verified": true
+      }
+      res.send(body)
+    },
+    
+    async send(req, res) {
+      let body = {
+        "sent": true
+      }
+      res.send(body)
+    },
+    
+    async delete(req, res) {
+      delete UserEmails[req.params.address]
+      Users[req.params.address]['emailExists'] = false
+      Users[req.params.address]['emailVerified'] = false
+      Users[req.params.address]['notificationsEnabled'] = false
```

Comment:
I think we should delete the sessions as well

Reply:
Hmm not sure, there are two separate buttons for log out and delete email, if we automatically logout when deleting email it will look like a bug.